test/multicall/Multicall.js

```diff
@@ -0,0 +1,328 @@
+    }
+
+    function getOutputLengthInBytes(res) {
+        let charSum = res.reduce(
+            (acc, value) => acc + value.length - 2, // do not count 0x
+            0
+        );
+
+        // 1 byte = 2 characters
+        return Math.floor(charSum / 2);
+    }
+
+    function getInputLengthInBytes(res) {
+        let charSum = res.reduce(
+            (acc, value) => acc + value.length,
+            0
+        );
+
+        // 1 byte = 1 character
+        return charSum;
+    }
+
+    before(async () => {
+        multicaller = await deployContract('Multicall3');
+        receiver = await deployContract('Receiver');
+        reverter = await deployContract('Reverter');
+    });
+
+    describe('static calls with large input', async function () {
+        it('should be able to aggregate 10 calls to processLongInput', async function () {
+            const n = 10;
+            const {callData, data} = prepareLongInputData(n, LONG_INPUT_ABI, LONG_INPUT_PARAMS);
+
+            const dataSize = getInputLengthInBytes(data);
+            expect(dataSize).to.be.eq(n * INPUT_ELEMENT_LENGTH);    // data is 2.6 kb
+
+            const res = await multicallProcessLongInput(callData);
+            expect(res).to.exist;
+            expect(res.length).to.eq(n);
+            for (let i = 0; i < n; i++) {
+                expect(res[i].success).to.eq(true)
+                expect(res[i].returnData).to.eq(RESULT_FIVE)
+            }
+        });
+
+        it('should be able to aggregate 1000 calls to processLongInput', async function () {
+            const n = 1000;
+            const {callData, data} = prepareLongInputData(n, LONG_INPUT_ABI, LONG_INPUT_PARAMS);
+
+            const dataSize = getInputLengthInBytes(data);
+            expect(dataSize).to.be.eq(n * INPUT_ELEMENT_LENGTH);    // data is 260 kb
+
+            const res = await multicallProcessLongInput(callData);
+            expect(res).to.exist;
+            expect(res.length).to.eq(n);
+            for (let i = 0; i < n; i++) {
+                expect(res[i].success).to.eq(true)
+                expect(res[i].returnData).to.eq(RESULT_FIVE)
+            }
+        });
+
+        it('should NOT be able to aggregate 5000 calls to processLongInput', async function () {
+            const n = 5000;
+            const {callData, data} = prepareLongInputData(n, LONG_INPUT_ABI, LONG_INPUT_PARAMS);
+
+            const dataSize = getInputLengthInBytes(data);
+            expect(dataSize).to.be.eq(1330000);    // data is 1,3 mb
+
+            // Input size is larger than 1 mb and the call is rejected by the relay
+            let hasError = false;
+            try {
+                const res = await multicallProcessLongInput(callData);
+            } catch (e) {
+                hasError = true;
+            }
+
+            expect(hasError).to.eq(true);
+        });
+
+
+        it('should be able to aggregate 11 calls to processLongInput and handles a revert', async function () {
```

Comment:
comment says 11 but constant below @211 is 10?

oh, i see, there's an internal detail of prepareLongInputData that's exposed here.  that's a bit tricky.